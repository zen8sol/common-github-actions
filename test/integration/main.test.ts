import {
  RunOptions,
  RunTarget,
  deleteAllFakedDirs,
} from "github-action-ts-run-api";
import { mkdirSync, readFileSync, rm, rmSync, writeFileSync } from "node:fs";
import path from "path";
import { afterAll, beforeEach, describe, expect, test } from "vitest";

const target = RunTarget.mainJs("action.yml");
const tmpDir = "./tmp";
const options = RunOptions.create()
  .setInputs({
    oldVersion: "1.0.1",
    newVersion: "1.1.0",
  })
  .setEnv({
    TODAY_DATE: "2023-12-02"
  })
  .setGithubContext({
    payload: {
      pull_request: { number: 123 },
      repository: { owner: { login: "test_owner" }, name: "test_repository" },
    },
  })
  .setTempDir(tmpDir)
  .setWorkspaceDir(`${tmpDir}/workspace`)
  .setFakeFsOptions({ rmFakedTempDirAfterRun: false, rmFakedWorkspaceDirAfterRun: false });

describe("main", () => {
  beforeEach(async () => {
    rmSync(tmpDir, { force: true, recursive: true });
    mkdirSync(`${tmpDir}/workspace`, {recursive:true});
  });

  test("should update changelog", async () => {
    const tmpChangelogFilePath = path.join('tmp/workspace', "CHANGELOG.md");

    writeFileSync(
      tmpChangelogFilePath,
      readFileSync("./test/resource/CHANGELOG.md")
    );
    const result = await target.run(options);

    expect(result.durationMs).lessThanOrEqual(1000);
    expect(result.stdout).toEqual("");
    expect(result.stderr).toEqual("");
    expect(result.commands.errors).toEqual([]);
    expect(result.runnerWarnings.length).toBe(0);
    expect(result.exitCode).toBe(0);

    const current = readFileSync(tmpChangelogFilePath).toString('utf-8');
    expect(current).toMatchSnapshot();
  });
});
