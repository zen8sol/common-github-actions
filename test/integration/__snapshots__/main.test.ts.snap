// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main > should update changelog 1`] = `
"# Changelog
ąęłó
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.1.0] - 2023-12-02

### Added

- added 1.

## [1.0.1] - 2023-11-29

### Added

- added 2.

### Fixed

- fixed 1

### Changed

- changed 1

### Removed

- removed 1

[unreleased] https://github.com/test_owner/test_repository/compare/1.1.0...HEAD   
[1.1.0] https://github.com/test_owner/test_repository/compare/0...1.1.0   
[1.0.1] https://github.com/test_owner/test_repository/releases/tag/1.0.1   
"
`;
